cypress/utils/dashboards/vis-augmenter/helpers.js

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { isEmpty } from 'lodash';
+import { MiscUtils } from '@opensearch-dashboards-test/opensearch-dashboards-test-library';
+
+const apiRequest = (url, method = 'POST', body = undefined, qs = undefined) =>
+  cy.request({
+    method: method,
+    failOnStatusCode: false,
+    url: url,
+    headers: {
+      'content-type': 'application/json',
+      'osd-xsrf': true,
+    },
+    body: body,
+    qs: qs,
+  });
+
+const devToolsRequest = (
+  url,
+  method = 'POST',
+  body = undefined,
+  qs = undefined
+) =>
+  cy.request({
+    method: 'POST',
+    form: false,
+    failOnStatusCode: false,
+    url: encodeURI(`api/console/proxy?path=${url}&method=${method}`),
+    headers: {
+      'content-type': 'application/json;charset=UTF-8',
+      'osd-xsrf': true,
+    },
+    body: body,
+    qs: qs,
+  });
+
+/**
+ * Cleans up the index & all associated saved objects (index pattern, visualizations,
+ * dashboards, etc.) created during the test run
+ */
+export const deleteVisAugmenterData = (
+  indexName,
+  indexPatternName,
+  visualizationNames,
+  dashboardName
+) => {
+  devToolsRequest(indexName, 'DELETE').then(() => {
+    apiRequest(
+      `api/saved_objects/index-pattern/${indexPatternName}`,
+      'DELETE'
+    ).then(() => {
+      apiRequest(
+        `
+api/opensearch-dashboards/management/saved_objects/_find?perPage=5000&page=1&fields=id&type=config&type=url&type=index-pattern&type=query&type=dashboard&type=visualization&type=visualization-visbuilder&type=augment-vis&type=search&sortField=type`,
+        'GET'
+      ).then((response) => {
+        response.body.saved_objects?.forEach((obj) => {
```

Comment:
Do you think we can do the following to pass the linter issues:

```js
if (!response.body.saved_objects) return;
response.body.saved_objects.forEach((obj) => {
```

Reply:
done